R/html_dependency.R

```diff
@@ -290,6 +291,12 @@ copyDependencyToDir <- function(dependency, outputDir, mustWork = TRUE) {
   } else dependency$name
   target_dir <- file.path(outputDir, target_dir)
 
+  # if overwrite is false check to see if the file already exists
+  if(!overwrite_dir & dir_exists(target_dir)){
```

Comment:
The condition `dir_exists(target_dir)` may be a little weak. A stronger version is to check if all filenames in the target dir and the dependency object are identical. This makes sure when the dependency is updated (in future versions of the package where the dependency is introduced), you'll get the new version.